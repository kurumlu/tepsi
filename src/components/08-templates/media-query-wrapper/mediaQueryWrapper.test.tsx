import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import {
  MediaQueryWrapper,
  MediaQueryWrapperProps,
  MediaQueryWrapperQuery,
} from './MediaQueryWrapper';

describe('<MediaQueryWrapper />', () => {
  it('renders <MediaQueryWrapper /> with default props', () => {
    const component: ShallowWrapper<MediaQueryWrapperProps> = shallow(
      <MediaQueryWrapper>Content</MediaQueryWrapper>
    );
    expect(component.prop('className')).toEqual(
      'sd-t-media-query-wrapper sd-t-media-query-wrapper--mobile-block'
    );
    expect(component).toMatchSnapshot('Default props');
  });

  it('renders <MediaQueryWrapper /> with desktop query prop', () => {
    const component: ShallowWrapper<MediaQueryWrapperProps> = shallow(
      <MediaQueryWrapper query={MediaQueryWrapperQuery.DESKTOP}>
        Content
      </MediaQueryWrapper>
    );
    expect(component.prop('className')).toEqual(
      'sd-t-media-query-wrapper sd-t-media-query-wrapper--desktop-block'
    );
    expect(component).toMatchSnapshot('desktop query prop');
  });

  it('renders <MediaQueryWrapper /> with tablet query prop and flex', () => {
    const component: ShallowWrapper<MediaQueryWrapperProps> = shallow(
      <MediaQueryWrapper query={MediaQueryWrapperQuery.TABLET} flex>
        Content
      </MediaQueryWrapper>
    );
    expect(component.prop('className')).toEqual(
      'sd-t-media-query-wrapper sd-t-media-query-wrapper--tablet-flex'
    );
    expect(component).toMatchSnapshot('tablet query prop and flex');
  });

  it('renders <MediaQueryWrapper /> with complex query prop', () => {
    const component: ShallowWrapper<MediaQueryWrapperProps> = shallow(
      <MediaQueryWrapper
        query={[
          MediaQueryWrapperQuery.MOBILE,
          MediaQueryWrapperQuery.TABLET_SMALL,
        ]}>
        Content
      </MediaQueryWrapper>
    );
    expect(component.prop('className')).toEqual(
      'sd-t-media-query-wrapper sd-t-media-query-wrapper--mobile-block sd-t-media-query-wrapper--tablet-small-block'
    );
    expect(component).toMatchSnapshot('complex query prop');
  });
});
