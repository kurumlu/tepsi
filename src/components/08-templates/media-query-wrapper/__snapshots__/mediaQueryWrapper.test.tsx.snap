// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaQueryWrapper /> renders <MediaQueryWrapper /> with complex query prop: complex query prop 1`] = `
<div
  className="sd-t-media-query-wrapper sd-t-media-query-wrapper--mobile-block sd-t-media-query-wrapper--tablet-small-block"
>
  Content
</div>
`;

exports[`<MediaQueryWrapper /> renders <MediaQueryWrapper /> with default props: Default props 1`] = `
<div
  className="sd-t-media-query-wrapper sd-t-media-query-wrapper--mobile-block"
>
  Content
</div>
`;

exports[`<MediaQueryWrapper /> renders <MediaQueryWrapper /> with desktop query prop: desktop query prop 1`] = `
<div
  className="sd-t-media-query-wrapper sd-t-media-query-wrapper--desktop-block"
>
  Content
</div>
`;

exports[`<MediaQueryWrapper /> renders <MediaQueryWrapper /> with tablet query prop and flex: tablet query prop and flex 1`] = `
<div
  className="sd-t-media-query-wrapper sd-t-media-query-wrapper--tablet-flex"
>
  Content
</div>
`;
